espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/ComposeNodeElement.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.model
+
+import android.graphics.Rect
+import androidx.compose.ui.semantics.*
+
+class ComposeNodeElement(private val node: SemanticsNode) {
+
+    val contentDescription: CharSequence? =
+        node.config.getOrNull(SemanticsProperties.ContentDescription)?.firstOrNull()
+
+    val text: CharSequence? = node.config.getOrNull(SemanticsProperties.Text)?.firstOrNull()
+
+    val resourceId: String = node.id.toString()
+
+    val viewTag: CharSequence? = node.config.getOrNull(SemanticsProperties.TestTag)
+
+    val isClickable: Boolean = node.config.contains(SemanticsActions.OnClick)
+
+    val isEnabled: Boolean = !node.config.contains(SemanticsProperties.Disabled)
+
+    val isFocused: Boolean = node.config.getOrNull(SemanticsProperties.Focused) == true
+
+    val isScrollable: Boolean = node.config.contains(SemanticsActions.ScrollBy)
+
+    val isSelected: Boolean = node.config.getOrNull(SemanticsProperties.Selected) == true
+
+    val className: String =
+        node.config.getOrNull(SemanticsProperties.Role)?.toString() ?: "ComposeNode"
+
+    val progress: String? =
+        node.config.getOrNull(SemanticsProperties.ProgressBarRangeInfo)?.toString()
```

Comment:
if an element has range info then we should assign its current value to `text` property. min and max value should also be represented as separate min/max attributes if exist

Reply:
I'd say for now lets only add text, because the other don't also exist in espresso